runtime_scan/pkg/provider/aws/config.go

```diff
@@ -15,40 +15,89 @@
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+)
 
 const (
-	AWSSubnetID            = "AWS_SUBNET_ID"
-	AWSJobImageID          = "AWS_JOB_IMAGE_ID"
-	AWSSecurityGroupID     = "AWS_SECURITY_GROUP_ID"
-	AWSInstanceType        = "AWS_INSTANCE_TYPE"
-	defaultAWSJobImageID   = "ami-0568773882d492fc8" // ubuntu server 22.04 LTS (HVM), SSD volume type
-	defaultAWSInstanceType = "t2.large"
+	EnvScannerRegion       = "VMCLARITY_AWS_SCANNER_REGION"
+	EnvSubnetID            = "VMCLARITY_AWS_SUBNET_ID"
+	EnvSecurityGroupID     = "VMCLARITY_AWS_SECURITY_GROUP_ID"
+	EnvKeyPairName         = "VMCLARITY_AWS_KEYPAIR_NAME"
+	EnvScannerImage        = "VMCLARITY_AWS_SCANNER_AMI_ID"
+	EnvScannerInstanceType = "VMCLARITY_AWS_SCANNER_INSTANCE_TYPE"
+	EnvBlockDeviceName     = "VMCLARITY_AWS_BLOCK_DEVICE_NAME"
+
+	DefaultScannerInstanceType = "t2.large"
+	DefaultBlockDeviceName     = "xvdh"
 )
 
 type Config struct {
-	AmiID           string // image id of a scanner job
-	SubnetID        string // the scanner's subnet ID
-	SecurityGroupID string // the scanner's security group
-	InstanceType    string // the scanner's instance type
+	// Region where the Scanner instance needs to be created
+	ScannerRegion string
+	// SubnetID where the Scanner instance needs to be created
+	SubnetID string
+	// SecurityGroupID which needs to be attached to the Scanner instance
+	SecurityGroupID string
+	// KeyPairName is the name of the SSH KeyPair to use for Scanner instance launch
+	KeyPairName string
+	// ScannerImage is the AMI image used for creating Scanner instance
+	ScannerImage string
+	// ScannerInstanceType is the instance type used for Scanner instance
+	ScannerInstanceType string
+	// BlockDeviceName contains the block device name used for attaching Scanner volume to the Scanner instance
+	BlockDeviceName string
 }
 
-func setConfigDefaults() {
-	viper.SetDefault(AWSJobImageID, defaultAWSJobImageID)
-	viper.SetDefault(AWSInstanceType, defaultAWSInstanceType)
+func (c *Config) Validate() error {
+	if c.ScannerRegion == "" {
+		return fmt.Errorf("parameter Region must not be nil")
+	}
+
+	if c.SubnetID == "" {
+		return fmt.Errorf("parameter SubnetID must not be nil")
+	}
+
+	if c.SecurityGroupID == "" {
+		return fmt.Errorf("parameter SecurityGroupID must not be nil")
+	}
+
+	if c.ScannerImage == "" {
+		return fmt.Errorf("parameter ScannerImage must not be nil")
+	}
+
+	if c.ScannerInstanceType == "" {
+		return fmt.Errorf("parameter ScannerInstanceType must not be nil")
+	}
 
-	viper.AutomaticEnv()
+	return nil
 }
 
-func LoadConfig() *Config {
-	setConfigDefaults()
+func NewConfig() (*Config, error) {
+	// Avoid modifying the global instance
+	v := viper.New()
+
+	v.AllowEmptyEnv(true)
+	v.AutomaticEnv()
+
+	_ = v.BindEnv("ScannerRegion", EnvScannerRegion)
+	_ = v.BindEnv("SubnetID", EnvSubnetID)
+	_ = v.BindEnv("SecurityGroupID", EnvSecurityGroupID)
+	_ = v.BindEnv("KeyPairName", EnvKeyPairName)
+	_ = v.BindEnv("ScannerImage", EnvScannerImage)
```

Comment:
not sure we want to be eating the errors here right? I'm not sure what the errors might be but it would probably be a good idea to handle them

Reply:
We are eating errors as `BindEnv` always returns a `nil` in this case.